### PR TITLE
Fix parser's dependency on ordered arguments

### PR DESCRIPTION
**Pull request by benchislett**

### CODE REVIEW

File: src/parser.cu [hunk 1]
```diff
@@ -480,15 +446,58 @@ Grid parseArgs(int argc, char** argv){
...
+    if (dimnum == 3) {
+        if (par.ival("yDim") == 1){
+            par.store("yDim",256);
+        }
+        if (par.ival("zDim") == 1){
+            par.store("zDim",256);
+        }
+    }
+
+    if (dimnum == 2 && par.ival("yDim") == 1){
```
Comment (benchislett) [inline src/parser.cu]: This was previously only executed when dimnum was explicitly set to 2, and not when it defaults to 2.
What exactly is this doing, and why?

Comment (leios) [reply src/parser.cu]: y and zDim are initially set to 1; however, if someone sets the dimensionality to be 2 or 3D, then we need to catch the case where they wanted higher dimensions, but forgot to explicitly set what they wanted for y and zDim. Here, we just set those to be default values of 256.

It's the same as the above clause, but for 2D.

Comment (benchislett) [reply src/parser.cu]: Wouldn't it be clearer to have all dimensions default to `256`, then change the unused ones to `1` afterwards?

Comment (leios) [reply src/parser.cu]: How do we check if they are unused vs the user explicitly setting the value to be `256`?

Comment (benchislett) [reply src/parser.cu]:
```If (dimnum < 3) {par.store("zDim", 1);}
if (dimnum < 2) {par.store("yDim", 1);}```

Comment (leios) [reply src/parser.cu]: Oh sure, that would probably work.

Comment (leios) [reply src/parser.cu]: This is in the `'c'` case, right? In that case, we have to be sure we can use either flag in any order

File: src/parser.cu [hunk 2]
```diff
@@ -78,6 +78,7 @@ Grid parseArgs(int argc, char** argv){
                 int xDim = atoi(optarg);
                 printf("Argument for x is given as %d\n",xDim);
                 par.store("xDim",(int)xDim);
+                break;
```
Comment (benchislett) [inline src/parser.cu]: For some reason, this line was removed in my last PR. I have no idea how, but it managed to slip through.

**Review by leios — APPROVED** [review]
This seems to work as intended.

Thanks for the PR

**Pull request merged**